app/controllers/concerns/foreman/controller/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,164 @@
+module Foreman::Controller::Puppet::HostsControllerExtensions
+  extend ActiveSupport::Concern
+
+  PUPPETMASTER_ACTIONS=[ :externalNodes, :lookup ]
+  PUPPET_AJAX_REQUESTS=%w{hostgroup_or_environment_selected puppetclass_parameters}
+
+  PUPPET_MULTIPLE_ACTIONS = %w(select_multiple_environment update_multiple_environment
+                               multiple_puppetrun update_multiple_puppetrun
+                               select_multiple_puppet_proxy update_multiple_puppet_proxy
+                               select_multiple_puppet_ca_proxy update_multiple_puppet_ca_proxy)
```

Comment:
I didn't delve too much into the code of the bulk actions, but I don't see too much in common between the actions. Maybe kind of helper mixin to ease the declaration of those, but definitely not a dedicated controller - bulk actions will be spread across multiple plugins. We can see it even in this PR - puppet actions have moved to the new mixin, but other ones like power and build stayed in the core.